_scripts/inject-protocol.sh

```diff
@@ -6,11 +6,51 @@ __file="${__dir}/$(basename "${BASH_SOURCE[0]}")"
 __base="$(basename ${__file} .sh)"
 source "${__dir}/_b3bp.sh"
 
-echo "--> Fetching latest protocol.."
-find _includes/tus.md -mtime +10 -exec rm -rf {} \; || true
-[ -f _includes/tus.md ] || (wget \
-  https://raw.githubusercontent.com/tus/tus-resumable-upload-protocol/master/protocol.md \
-  --quiet \
-  --output-document=_includes/tus.md \
-  && echo "$(tail -n +3 ./_includes/tus.md)" > ./_includes/tus.md)
-# Removes first two lines to allow our own h2 header
+function fetch_protocol {
+  local url=$1
+  local slug=$2
+  local outdated=$3
+
+  local settings=""
+
+  # The pages of old protocol versions should not contains comments
+  # and should not be indexed by search engines.
+  if [ "${outdated}" == "1" ]
+  then
+    settings="
+noindex: true
+permalink: /protocols/resumable-upload/:slug.html
+version_outdated: true"
+  else
+    settings="
+comments: true
+permalink: /protocols/resumable-upload.html
+version_outdated: false"
+  fi
+
+  echo "--> Fetching protocol ${slug}"
+
+  # Removes first two lines to allow our own h2 header
+  local content="$(curl --silent ${url} | tail -n +3)"
```

Comment:
does shellcheck not complaint that you should move the $() to a different line than the declaration, or any error is swallowed?

may also want to add --fail and --retry 3 to curl

Reply:
Good call!